src/main/java/maksym/fedorenko/bookstore/controller/CategoryController.java

```diff
@@ -0,0 +1,101 @@
+package maksym.fedorenko.bookstore.controller;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import jakarta.validation.constraints.Positive;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import maksym.fedorenko.bookstore.dto.book.BookDtoWithoutCategories;
+import maksym.fedorenko.bookstore.dto.category.CategoryDto;
+import maksym.fedorenko.bookstore.dto.category.CreateCategoryRequestDto;
+import maksym.fedorenko.bookstore.dto.category.UpdateCategoryRequestDto;
+import maksym.fedorenko.bookstore.service.CategoryService;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "Category management", description = "Endpoints for mapping categories")
+@RestController
+@RequestMapping("/api/categories")
```

Comment:
It looks strange that you use the prefix `api` in each controller but for now, you can leave it.